samza-core/src/main/scala/org/apache/samza/config/StorageConfig.scala

```diff
@@ -38,7 +39,23 @@ class StorageConfig(config: Config) extends ScalaMapConfig(config) with Logging
   def getStorageFactoryClassName(name: String) = getOption(FACTORY.format(name))
   def getStorageKeySerde(name: String) = getOption(StorageConfig.KEY_SERDE format name)
   def getStorageMsgSerde(name: String) = getOption(StorageConfig.MSG_SERDE format name)
-  def getChangelogStream(name: String) = getOption(CHANGELOG_STREAM format name)
+  def getChangelogStream(name: String) = {
+    // If the config specifies 'stores.<storename>.changelog' as '<system>.<stream>' combination - it will take precedence.
+    // If this config only specifies <astream> and there is a value in job.changelog.system=<asystem> -
+    // these values will be combined into <asystem>.<astream>
+    val systemStream = getOption(CHANGELOG_STREAM format name)
+    val changelogSystem = getOption(CHANGELOG_SYSTEM)
+    val systemStreamRes = if ( systemStream.isDefined
+            && ! systemStream.getOrElse("").contains('.') // contains only stream name
+            && changelogSystem.isDefined) {
+      // get the system name
+      Some(changelogSystem.get + "." + systemStream)
+    } else {
```

Comment:
Shouldn't it error out if systemStream only has stream name and changelogSystem is not defined?